src/status_im/contexts/wallet/send/events.cljs

```diff
@@ -338,32 +344,42 @@
+                                                                      :only-with-balance? false}))
          token-networks-ids (when token (mapv #(:chain-id %) (:networks token)))
-         sender-network-values (when token-available-networks-for-suggested-routes
+         sender-network-values (when sender-token-available-networks-for-suggested-routes
                                  (send-utils/loading-network-amounts
-                                  {:valid-networks     (if (= transaction-type :tx/bridge)
-                                                         (filter
-                                                          #(not= bridge-to-chain-id %)
-                                                          token-available-networks-for-suggested-routes)
-                                                         token-available-networks-for-suggested-routes)
+                                  {:valid-networks
+                                   (if (= transaction-type :tx/bridge)
+                                     (filter
+                                      #(not= bridge-to-chain-id %)
+                                      sender-token-available-networks-for-suggested-routes)
+                                     sender-token-available-networks-for-suggested-routes)
                                    :disabled-chain-ids disabled-from-chain-ids
-                                   :receiver-networks  receiver-networks
+                                   :receiver-networks receiver-networks
                                    :token-networks-ids token-networks-ids
-                                   :tx-type            transaction-type
-                                   :receiver?          false}))
-         receiver-network-values (when token-available-networks-for-suggested-routes
+                                   :tx-type transaction-type
+                                   :receiver? false}))
+         receiver-network-values (when receiver-token-available-networks-for-suggested-routes
                                    (send-utils/loading-network-amounts
-                                    {:valid-networks (if (= transaction-type :tx/bridge)
-                                                       (filter
-                                                        #(= bridge-to-chain-id %)
-                                                        token-available-networks-for-suggested-routes)
-                                                       token-available-networks-for-suggested-routes)
+                                    {:valid-networks
+                                     (if (= transaction-type :tx/bridge)
+                                       (filter
+                                        #(= bridge-to-chain-id %)
+                                        receiver-token-available-networks-for-suggested-routes)
+                                       receiver-token-available-networks-for-suggested-routes)
```

Comment:
This function/event looks very complex, it'd be great if we solve this problem in smaller functions with meaninful names, I know this is out of scope of this PR, but IMO this function needs to be refactored to be easier to understand and maintain.

Reply:
Agree, we should even write tests for these events. Will file an issue for it.

Reply:
Created #20162